databroker/_core.py

```diff
@@ -2145,6 +2144,22 @@ def from_config(cls, config, auto_register=True):
         for spec, handler in config.get('handlers', {}).items():
             cls = load_cls(handler)
             db.reg.register_handler(spec, cls)
+        # if 'root_map' in config, set the root_map
+        if 'root_map' in config:
+            root_map = config['root_map']
+            if not isinstance(root_map, dict):
+                print("root_map is not right type in config file. "
+                      "It must be tuple of length 2 (see documentation)")
+                raise TypeError
+            # TODO : Allow multiple mappings, need to update
+            # db.reg.set_root_map first
+            if len(root_map) != 1:
```

Comment:
Why this restriction?    I think this entire block can just be `db.reg.set_rootmap(root_map)`